src/utils/ExplorerUtil.ts

```diff
@@ -107,20 +108,24 @@ export const ExplorerUtil = {
       const nativeUrl = CoreUtil.formatNativeUrl(deepLink, wcURI);
       const universalUrl = CoreUtil.formatUniversalUrl(universalLink, wcURI);
       if (nativeUrl) {
+        StorageUtil.setDeepLinkWallet(deepLink!);
         await Linking.openURL(nativeUrl).catch(() => {
           // Fallback to universal link
           if (universalUrl) {
             Linking.openURL(universalUrl);
+            StorageUtil.setDeepLinkWallet(universalLink!);
```

Comment:
Perhaps add `universalLink` to the if statement that checks `universalurl` to get rid of `!` here as well?